src/functions_framework/event_conversion.py

```diff
@@ -168,3 +179,50 @@ def _split_resource(context: Context) -> Tuple[str, str, str]:
         raise EventConversionException("Resource regex did not match")
 
     return service, match.group(1), match.group(2)
+
+
+def marshal_background_event_data(request):
+    """Marshal the request body of a raw Pub/Sub HTTP request into the schema that is expected of
+    a background event"""
+    request_data = request.get_json()
+    if not _is_raw_pubsub_payload(request_data):
+        # If this in not a raw Pub/Sub request, return the unaltered request data.
+        return request_data
+    
+    return {
+        "context": {
+            "eventId": request_data["message"]["messageId"],
+            "timestamp": datetime.utcnow().isoformat() + "Z",
+            "eventType": _PUBSUB_EVENT_TYPE,
+            "resource": {
+                "service": _PUBSUB_CE_SERVICE,
+                "type": _PUBSUB_MESSAGE_TYPE,
+                "name": _parse_pubsub_topic(request.path),
+            },
+        },
+        "data": {
+            '@type': _PUBSUB_MESSAGE_TYPE,
+            "data": request_data["message"]["data"],
+            "attributes": request_data["message"]["attributes"],
+        }
+    }
```

Comment:
We may want to wrap this is a try/except for a `KeyError` or otherwise guard against potentially hand-crafted requests that are missing `messageId`, `data` or `attributes` values on the `message` and raise a `EventConversionException` instead.